File: java/parsers/src/test/java/jp/co/teruuu/parser/common/PlusParserTest.java
package jp.co.teruuu.parser.common;

import jp.co.teruuu.parser.common.ParseResult;
import jp.co.teruuu.parser.common.Parser;
import org.junit.Test;

import java.util.Arrays;
import java.util.List;

public class PlusParserTest {

    @Test
    public void test() {
        Parser<List<String>> manyParser = Parser.string("Hello").plus();

        ParseResult result1 = manyParser.parse("HelloHelloHello");
        assert(result1 instanceof ParseResult.Success);
        assert(((ParseResult.Success) result1).value.equals(Arrays.asList("Hello", "Hello", "Hello")));

        ParseResult result2 = manyParser.parse("Hello HelloHello");
        assert(result2 instanceof ParseResult.Success);
        assert(((ParseResult.Success) result2).value.equals(Arrays.asList("Hello")));

        ParseResult result3 = manyParser.parse("helloHelloHello");
        assert(result3 instanceof ParseResult.Failure);
    }
}

File: scala/parsers/src/test/scala/jp/co/teruuu/parser/common/PlusParserTest.scala
package jp.co.teruuu.parser.common

import org.scalatest.FlatSpec;

class PlusParserTest extends FlatSpec {

  "parse" should "successt" in {
    val parser = Parser.string("Hello").plus()
    val result1 = parser.parse("HelloHello")
    assert(result1.isInstanceOf[ParseOk[_]])
    assert(result1.asInstanceOf[ParseOk[List[String]]].value.equals(List("Hello", "Hello")))

    val result2 = parser.parse("hello")
    assert(result2.isInstanceOf[ParseNg[_]])
  }

}
